src/libstore/derivations.hh

```diff
@@ -136,11 +136,15 @@ struct DerivationOutput : _DerivationOutputRaw
         const Store & store,
         std::string_view drvName,
         std::string_view outputName) const;
+    /**
+     * @param xpSettings Stop-gap to avoid globals during unit tests.
```

Comment:
Testability is important enough to warrant a parameter like this, but it is transitively a stop-gap, because experimental features are a stop-gap.